src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -2223,6 +2229,101 @@ ssize_t HwEmShim::xclUnmgdPread(unsigned flags, void *buf, size_t count, uint64_
   return xclCopyBufferDevice2Host(buf, offset, count, 0 , 0);
 }
 
+
```

Comment:
Was there a reason why this entire section was added to shim.cxx instead of perf.cxx? Moving it to perf.cxx would be a better place so all debug/profile code gets isolated and shim.cxx is not overloaded with it.

Reply:
With the new ProfileIP class based flow in re-architected XDP, the reading of profile/trace data from monitors is handled inside XDP using xclRead/Write and only a couple of flow dependent Shim methods (added in the current section). When this support is fully verified, the older implementation of profiling can be removed entirely. We may not need a separate file for profiling then and perf.cxx may be removed. That is why, I have added these 3 new SHIM methods for profiling in shim.cxx.

Reply:
Okay, makes sense.